glide-core/src/client/mod.rs

```diff
@@ -137,12 +136,14 @@ async fn run_with_timeout<T>(
         Some(duration) => match tokio::time::timeout(duration, future).await {
             Ok(result) => result,
             Err(_) => {
-                // Record timeout error metric
-                if let Err(e) = GlideOpenTelemetry::record_timeout_error() {
-                    log_error(
-                        "OpenTelemetry:timeout_error",
-                        format!("Failed to record timeout error: {}", e),
-                    );
+                // Record timeout error metric if telemetry is initialized
+                if GlideOpenTelemetry::is_initialized() {
```

Comment:
so each time we're going to record a metric we're going to call this function?
If so, I think it is better to do the is_init check inside the "record_*" functions, and add description that if OTEL isn't initialized it would do nothing